src/api/app/models/project.rb

```diff
@@ -1257,12 +1257,14 @@ def bsrequest_repos_map(project)
 
   private :bsrequest_repos_map
 
+  # NOTE: This has to cover project name validations in backend/BSVerify.pm (verify_projid)
   def self.valid_name?(name)
     return false unless name.is_a?(String)
     return false if name == '0'
-    return false if name =~ /::/
+    return false if name =~ /:[:\._]/
+    return false if name =~ /\A[:\._]/
     return false if name.end_with?(':')
-    return true if name =~ /\A[a-zA-Z0-9][-+\w\.:]*\z/
+    return true  if name =~ /\A[\w\-+\.:]+\z/
```

Comment:
Nitpick: '-' was not escaped before. Does it need to be escaped?

Reply:
Only when you use the regex '-', which I did in one of my attempts to change the regex. I'll fix it.